tools/launcher/desktop_main.cpp

```diff
@@ -130,25 +151,47 @@ int main(int argc, const char** argv)
 
             if (success == true)
             {
-                STARTUPINFOA si = {};
-                si.cb           = sizeof(si);
-
-                PROCESS_INFORMATION pi = {};
-
-                std::string interceptor_path = gfxrecon::util::interception::GetInterceptorPath(process_info.app_path);
-
-                gfxrecon::util::interception::LaunchAndInjectA(
-                    process_info.app_path.c_str(),
-                    const_cast<LPSTR>(process_info.app_path_plus_args.c_str()),
-                    nullptr,
-                    nullptr,
-                    TRUE,
-                    CREATE_DEFAULT_ERROR_MODE | CREATE_SUSPENDED,
-                    nullptr,
-                    process_info.app_dir.c_str(),
-                    &si,
-                    &pi,
-                    interceptor_path.c_str());
+                char working_dir[MAX_PATH] = {};
+                GetCurrentDirectory(MAX_PATH, working_dir);
+
+                // Save off GFXR location in an environment variable, which is read later as our libs are loaded in
+                SetEnvironmentVariableA(gfxrecon::util::interception::kGfxrInstallEnv, working_dir);
+
+                // If we're using the launcher, we can assume detours will be used
+                SetEnvironmentVariableA("GFXRECON_USE_DETOURS_HOOKING", "true");
```

Comment:
Could you add some error handling in case `GetCurrentDirectory` and `SetEnvironmentVariableA` fail? Maybe just check for success and log a warning/error message if not.

Reply:
fixed